graf3d/eve7/src/REveTypes.cxx

```diff
@@ -40,3 +40,20 @@ REveException REX::operator+(const REveException &s1,  const char *s2)
 
 REveException REX::operator+(const REveException &s1, ElementId_t x)
 { REveException r(s1); r.append(std::to_string(x)); return r; }
+
+////////////////////////////////////////////////////////////////////////////////
+
+thread_local REX::REveLog REX::gEveLog;
+
+void REveLog::add(const char* txt) { fLog << txt; std::cout << txt << std::endl;}
+void REveLog::add(const std::string& txt) { fLog << txt; std::cout << txt << std::endl;}
+bool REveLog::has_contents() { return ! fLog.str().empty(); }
+void REveLog::clear() { fLog.clear(); }
+
+
+REveLog &REveLog::operator<<(std::ostream &(*os)(std::ostream &))
+{
+   fLog << os;
+   std::cout << os;
+   return *this;
+}
```

Comment:
missing newline

Reply:
@linev I will replace REveLog with RLogger in a new PR. We introduce REveLog because we wanted to send the server messages to client and then empty the log: https://github.com/alja/root/blob/server-threads/graf3d/eve7/src/REveManager.cxx#L922
I guess that is possible also in RLogger.

Reply:
I mean one should use RLogger only for that part which appears on the standard output.